test/e2e/util.go

```diff
@@ -1145,7 +1149,7 @@ func RunRC(config RCConfig) error {
 			dumpNodeDebugInfo(config.Client, containerRestartNodes.List())
 			return fmt.Errorf("%d containers failed which is more than allowed %d", failedContainers, maxContainerFailures)
 		}
-		if len(pods) < len(oldPods) || len(pods) > config.Replicas {
+		if len(pods) < len(oldPods) {
 			// This failure mode includes:
```

Comment:
Are both failure cases still relevant in this failure?  I think the first is (is accounts for the pod count drop), but the 2nd might be more applicable to the overshoot case.